src/windshaft-integration/model-updater.js

```diff
@@ -266,4 +268,22 @@ ModelUpdater.prototype._getLayerLegends = function (layerModel) {
     layerModel.legends.choropleth
   ];
 };
+
+/**
+ * Return the analysis list without duplicates
+ *
+ * (a dataview and a layer could share some analysis)
+ */
+ModelUpdater.prototype._getUniqueAnalyses = function () {
```

Comment:
I'd include "nodes" in the name of this method somehow so that it's clear that it's not just returning "root" analyses but it's "flattening" the analysis trees instead.

Reply:
👍